components/concurrency_manager/src/lock_table.rs

```diff
@@ -21,13 +21,19 @@ impl Default for LockTable {
 impl LockTable {
     pub async fn lock_key(&self, key: &Key) -> KeyHandleGuard {
         loop {
-            let handle = Arc::new(KeyHandle::new(key.clone(), self.clone()));
+            let handle = Arc::new(KeyHandle::new(key.clone()));
```

Comment:
We could add some comments to demonstrate that this handle object's relationship with the `LockTable` like that it may or may not be the used one.